akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/AccountExampleOneLiners.java

```diff
@@ -0,0 +1,157 @@
+            this.amount = amount;
+        }
+    }
+    public static class Withdrawn implements AccountEvent {
+        public final double amount;
+
+        Withdrawn(double amount) {
+            this.amount = amount;
+        }
+    }
+    public static class AccountClosed implements AccountEvent {}
+
+    interface Account {}
+    public static class EmptyAccount implements Account {}
+    public static class OpenedAccount implements Account {
+        public final double balance;
+
+        OpenedAccount(double balance) {
+            this.balance = balance;
+        }
+    }
+    public static class ClosedAccount implements Account {}
+
+    public static Behavior<AccountCommand> behavior(String accountNumber) {
+        return Behaviors.setup(context -> new AccountExampleOneLiners(context, accountNumber));
+    }
+
+    public AccountExampleOneLiners(ActorContext<AccountCommand> context, String accountNumber) {
+        super(accountNumber);
+    }
+
+    @Override
+    public Account emptyState() {
+        return new EmptyAccount();
+    }
+
+    private Effect<AccountEvent, Account> createAccount() {
+        return Effect().persist(new AccountCreated());
+    }
+
+    private Effect<AccountEvent, Account> depositCommand(Deposit deposit) {
+        return Effect().persist(new Deposited(deposit.amount));
+    }
+
+    private Effect<AccountEvent, Account> withdrawCommand(OpenedAccount account, Withdraw withdraw) {
+        if ((account.balance - withdraw.amount) < 0.0) {
+            return Effect().unhandled(); // TODO replies are missing in this example
+        } else {
+            return Effect().persist(new Withdrawn(withdraw.amount))
+                    .andThen(acc2 -> { // FIXME in scaladsl it's named thenRun, change javadsl also?
+                        // we know this cast is safe, but somewhat ugly
+                        OpenedAccount openAccount = (OpenedAccount) acc2;
+                        // do some side-effect using balance
+                        System.out.println(openAccount.balance);
+                    });
+        }
+    }
+
+    private Effect<AccountEvent, Account> closeCommand(OpenedAccount account, CloseAccount cmd) {
+        if (account.balance == 0.0)
+            return Effect().persist(new AccountClosed());
+        else
+            return Effect().unhandled();
+    }
+
+
+    private CommandHandlerBuilder<AccountCommand, AccountEvent, EmptyAccount, Account> initialHandler() {
+        return commandHandlerBuilder(EmptyAccount.class)
+                .matchCommand(CreateAccount.class, this::createAccount);
+    }
+
+    private CommandHandlerBuilder<AccountCommand, AccountEvent, OpenedAccount, Account> openedAccountHandler() {
+        return commandHandlerBuilder(OpenedAccount.class)
+                .matchCommand(Deposit.class, this::depositCommand)
+                .matchCommand(Withdraw.class, this::withdrawCommand)
+                .matchCommand(CloseAccount.class, this::closeCommand);
```

Comment:
@renatocaval another you can try is to start writing here
```
.matchCommand(Withdraw.class, this::anotherWithdraw)
```

and then alt-enter on the red anotherWithdraw and let IntelliJ generate the method. Does that work also with the additional overloaded matches? (I found that workflow pretty neat)

Reply:
If I understand you correctly, you are suggesting to not have overloaded method because IntelliJ would generate it for the user. Correct?

I tried it, and it's the variation accepting a `Supplier` that is breaking it. 

When we have:
```scala
def matchCommand[C <: Command](commandClass: Class[C], handler: Supplier[Effect[Event, State]])
```
IntelliJ generates: 
```java
private Effect<AccountEvent, Account> anotherCommand()
```

If I remove it, IntelliJ generates: 
```java
private Effect<AccountEvent, Account> anotherCommand(OpenedAccount account, AccountCommand accountCommand) 
```

I can remove the State if I don't need it, for instance, the command handler for deposit doesn't need it.

That works in all examples I have in this PR, but for the examples with handlers in-model, IntelliJ adds a static method. 

I guess we should remove the Supplier variant. This will also break in the face of `ReplyEffect`. We can't make an effect without having a Cmd, so we can't make one out of thin air.